### PR TITLE
test: add `buildHeaders` function for http sync tests

### PR DESCRIPTION
**Pull request by ChihweiLHBird**
## This PR

The previous way of building HTTP headers with raw map is not reliable in some edge cases.

An example is case-sensitive issue for `ETag` header which requires `Etag` (lower case `t`) to be the map key when building the response with a raw map, in order to receive it via `resp.Header.Get("ETag")` on the other end.

Having a `buildHeaders` function to utilize `Set` built-in function of `http.Header` would solve this concern.

### How to test
```bash
go test ./core/pkg/sync/http/...
```

### CODE REVIEW

**Review by aepfli — APPROVED** [review]
Thank you, this is a great improvement. For me this change is fine as is, but i would highly appreciate your feedback/insights on gemini's suggestion

**Comment by ChihweiLHBird** [comment]
@aepfli Thanks for the review! I think the suggestion from Gemini is great but I would prefer a more structured way to handle the issue it pointed out. I just updated the function to accept a `map[string][]string` parameter and uses `h.Add(...)` function to support headers with multiple values. Do you think this is fine? I am also open to adopt the Gemini's way (variadic string) if you like it.

File: core/pkg/sync/http/http_sync_test.go [hunk 1]
```diff
@@ -16,6 +16,14 @@ import (
 	"go.uber.org/mock/gomock"
 )
 
+func buildHeaders(m map[string]string) http.Header {
+	h := http.Header{}
+	for k, v := range m {
+		h.Set(k, v)
+	}
+	return h
+}
```
[the remark this answers is not in the document]
Comment (ChihweiLHBird) [reply core/pkg/sync/http/http_sync_test.go]: Concern addressed in https://github.com/open-feature/flagd/pull/1804/commits/0666c44664312a47079d7d36e29d114d312bb87b

**Comment by aepfli** [comment]
> @aepfli Thanks for the review! I think the suggestion from Gemini is great but I would prefer a more structured way to handle the issue it pointed out. I just updated the function to accept a `map[string][]string` parameter and uses `h.Add(...)` function to support headers with multiple values. Do you think this is fine? I am also open to adopt the Gemini's way (variadic string) if you like it.

I like your approach, i would stick with yours.

**Review by chrfwow — APPROVED** [review]

**Review by beeme1mr — APPROVED** [review]
